pkg/prometheus/promcfg.go

```diff
@@ -2281,6 +2282,173 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 			Value: configs,
 		})
 	}
+
+	//ConsulSDConfig
+	if len(sc.Spec.ConsulSDConfigs) > 0 {
+		configs := make([][]yaml.MapItem, len(sc.Spec.ConsulSDConfigs))
+		for i, config := range sc.Spec.ConsulSDConfigs {
+			configs[i] = cg.addBasicAuthToYaml(configs[i], fmt.Sprintf("scrapeconfig/%s/%s/consulsdconfig/%d", sc.Namespace, sc.Name, i), store, config.BasicAuth)
```

Comment:
before referencing the assets from the store, we need to validate and load them in `*ResourceSelector.SelectScrapeConfigs()`.

Reply:
Can you take a look again @simonpasquier?
I'm not sure if I changed on the right place